### PR TITLE
Concourse AL2

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
Config to build AL2 Concourse AMI

### CODE REVIEW

File: dw-al2-concourse-ami/dw-concourse-ami-install.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+sed -i -e 's/selinux=0/selinux=1 enforcing=0/' /boot/grub/menu.lst
+
+# Relax umask settings and defaults
+sed -i 's/^.*umask 0.*$/umask 002/' /etc/bashrc
+sed -i 's/^.*umask 0.*$/umask 002/' /etc/profile
+sed -i 's/^.*umask 0.*$/umask 002/' /etc/profile.d/*.sh
+sed -i 's/^umask 027/umask 002/' /etc/init.d/functions
+
+# Download and Install Concourse
+concourse_version=6.1.0
```
Comment (mattburgess) [inline dw-al2-concourse-ami/dw-concourse-ami-install.sh:28]: Let's install the right version here please. I know we download and install the right version during userdata but that's a horrible pattern that we should fix during this work.

An upgrade will mean a new AMI, but that's fine, IMO; software should be baked in to the AMI, not installed during instance startup.

File: dw-al2-concourse-ami/generic_packer_template.json.j2 [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    },
+    "builders": [{
+      "type": "amazon-ebs",
+      {% if 'source_ami_id' in event and event['source_ami_id']|length > 0 %}
+        "source_ami": "{{ event['source_ami_id']}}",
+      {% endif %}
+      "source_ami_filter": {
+        "filters": {
+          "virtualization-type": "{{ event['source_ami_virt_type'] or 'hvm' }}",
+          "name": "{{ event['source_ami_name'] or 'amzn-ami-hvm-*' }}",
```
Comment (mattburgess) [inline dw-al2-concourse-ami/generic_packer_template.json.j2:15]:
```suggestion
          "name": "{{ event['source_ami_name'] or 'amzn2-ami-hvm-*' }}",
```
Just a safeguard such that if don't pass in a source AMI at least you'll get an AL2 one back.

File: dw-al2-concourse-ami/dw-concourse-ami-install.sh (hunk 1 above)
Comment (ChrisScottThomas) [reply dw-al2-concourse-ami/dw-concourse-ami-install.sh:28]: saw it and fixed.

File: dw-al2-concourse-ami/generic_packer_template.json.j2 (hunk 2 above)
Comment (ChrisScottThomas) [reply dw-al2-concourse-ami/generic_packer_template.json.j2]: Ah, thanks.

**Review by mattburgess — APPROVED** [review]

**Review by steveburton4 — APPROVED** [review]

**Pull request merged**